rfcs/0128-selective-auto-merge-on-bot-upgrades.md

```diff
@@ -0,0 +1,53 @@
+---
+feature: Selective auto-merge on bot upgrades
+start-date: 2022-07-07
+author: superherointj
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a `meta.autoMerge` attribute to packages to allow committers to delegate merge rights to package maintainers for bot auto updates.
+
+# Motivation
+[motivation]: #motivation
+
+* Reduce pending PRs for review and merge in nixpkgs.
+* Save reviewers and commiters time.
+* Speed up package upgrades.
+
+# Detailed design
+[design]: #detailed-design
+
+Add a new `meta.autoMerge` package attribute with type `bool` defaulting to `false`. To be documented in Nixpkgs manual.
+
+OfBorg reads `meta.autoMerge` value and sets `13.auto-merge` label to PR.
+
+GitHub Actions will trigger a merge when the following 4 conditions are met:
+1) `@r-ryantm` is PR's author.
+2) Has `13.auto-merge` label.
+3) All CI checks passed.
+4) A package maintainer has approved PR. (Label: `12.approved-by: package-maintainer`)
```

Comment:
Same as here. I don't think we can rely in labels for this feature. It would open for any maintainer to merge any `@r-ryantm` PRs by simply setting two labels.

Unless of course, I am wrong about the `nixpkgs-maintainers` being able to set labels thing.

Reply:
> Same as here. I don't think we can rely in labels for this feature. 

Labels are only to be used to trigger the GitHub Action which is going to re-check rules. (From a trusted source.)

> It would open for any maintainer to merge any `@r-ryantm` PRs by simply setting two labels.

No.

> Unless of course, I am wrong about the `nixpkgs-maintainers` being able to set labels thing.

Yes, maintainers can set labels.